qoajad_backend/src/main/java/com/qoajad/backend/controller/DefaultController.java

```diff
@@ -0,0 +1,52 @@
+package com.qoajad.backend.controller;
+
+import com.qoajad.backend.model.User;
+import com.qoajad.backend.service.user.UserService;
+import org.bouncycastle.cert.ocsp.Req;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.cloud.context.config.annotation.RefreshScope;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.util.List;
+
+@RefreshScope
+@RestController
+public class DefaultController {
+
+    private final UserService userService;
+
+    @Autowired
+    public DefaultController(@Qualifier("defaultUserService") final UserService userService) {
+        this.userService = userService;
+    }
+
+    @RequestMapping(value = "/users", method = RequestMethod.GET)
+    public ResponseEntity<List<User>> retrieveAllUsers() {
+        return userService.retrieveAllUsers();
+    }
+
+    @RequestMapping(value = "/user/{id}", method = RequestMethod.GET)
+    public ResponseEntity<User> readUser(@PathVariable Integer id) {
```

Comment:
Ditto.

Reply:
Btw, consider passing from this point and downstream code an int instead of an Integer. An Integer is a class that contains an int, Hence it is heavier to deal with (Has many methods that we don't require) We also don't need to include this to be processed by the garbage collection later, instead what you can do is change the downstream code of the signature for function parameters from Integer to int.